File: main/src/main/java/net/fwitz/math/main/binary/complex/analysis/zeta/ZetaPlot.java
package net.fwitz.math.main.binary.complex.analysis.zeta;

import net.fwitz.math.binary.complex.analysis.RiemannZeta;
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot;
import net.fwitz.math.plot.binary.complex.color.DomainColoringContour;

public class ZetaPlot {
    public static void main(String[] args) {
        new ComplexFunctionPlot("zeta(s)")
                .size(500, 1000)
                .domainX(-10, 10)
                .domainY(-20, 20)
                .computeFn(RiemannZeta::zeta)
                //.colorFn(new DomainColoringZeroBasin())
                .colorFn(new DomainColoringContour())
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/complex/analysis/zeta/ZetaPlot.kt
package net.fwitz.math.main.binary.complex.analysis.zeta

import net.fwitz.math.binary.complex.analysis.RiemannZeta
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot
import net.fwitz.math.plot.binary.complex.color.DomainColoringContour
import net.fwitz.math.plot.binary.complex.color.DomainColoringZeroBasin

object ZetaPlot {
    @JvmStatic
    fun main(args: Array<String>) = ComplexFunctionPlot("zeta(s)")
        .size(500, 500)
        .domainX(-20.0, 20.0)
        .domainY(-20.0, 20.0)
        .computeFn(RiemannZeta::zeta)
        .colorFn(DomainColoringContour)
        .render()
}